backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/order/products_proposal/model/ProductsProposalRowsLoader.java

```diff
@@ -224,12 +226,16 @@ private ProductsProposalRow toProductsProposalRowOrNull(@NonNull final I_M_Produ
 				? packingMaterialsService.getDisplayName(packingMaterialId)
 				: TranslatableStrings.empty();
 
+		final Predicate<OrderLine> asiMatcher = orderLine -> attributeSetInstanceBL.asiValuesMatchOrEmpty(orderLine.getAsiId(), extractProductASI(record));
```

Comment:
@metas-rc @adi-stefan auch, now i see that we will load the record's ASI on each call.
IMHO asiValuesMatchOrEmpty method shall be changed to work with `ImmutableAttributeSet` which is loaded only once here.